tasks/mysql_secure_installation.yml

```diff
@@ -22,6 +22,7 @@
     - '::1'
     - '127.0.0.1'
     - 'localhost'
+  when: mysql_check_root_password_presence
 
 - name: install .my.cnf with credentials
```

Comment:
if you set mysql_check_root_password_presence to _false_ and have different password installed - all further steps related to DB changes will fail anyway, because in this case you don't execute previous taks, also my.cnf and actual root password will be inconsistent. 
I'd say it even creates a high risk of losing a password